### DISCUSSION TITLE
Calling Rust from Slint / Callbacks

### DISCUSSION THREAD

**Discussion by darkwyrm** [slint-ui/slint#1833] [Q&A] [answered]
I'm still new to figuring all this out, so pardon me if this is ridiculously simple to do.

I have a ListView and a TextEdit such that I'd like to be able to click on an item in the list, have it call some backend Rust code to load text from a database, and drop it into the TextEdit. I've looked through the repo and the documentation and can't figure out where to begin to get this working. How do I do this?

**Comment by ogoffart** [comment, marked as answer]
The way to do that in Slint is with callback.
For example, you can put a callback on a global object, and assogn an handler to it.
See https://docs.rs/slint/latest/slint/docs/recipes/index.html#global-callbacks

You can then do something like
```
clicked => { 
  GlobalLogic.item_was_clicked(item_index, some_other_data);
}
```

**Reply by darkwyrm** [reply]
That makes sense now. Thank you!
